articles/quickstart/spa/vuejs/_includes/_centralized_login.md

```diff
@@ -2,475 +2,441 @@
-        if (this.handleLoginEvent) {
-          authService.removeListener('loginEvent', this.handleLoginEvent);
-        }
-      }
-    });
-  }
-};
+```json
+{
+  "domain": "${account.namespace}",
+  "clientId": "${account.clientId}"
+}
 ```
 
-This plugin provides access to the `AuthService` class from each component, through the `this.$auth` property. It also provides a mechanism for when the login state changes, for components that implement a `handleLoginEvent` method.
+:::note
+This configuration file contains values relating to your Auth0 app, and should not be committed into source control.
+:::
+
+<!-- ![hosted login](/media/articles/web/hosted-login.png) -->
 
-Open `main.js` and install the plugin:
+Finally, open `src/main.js` and use `Vue.use` to install the plugin. Notice that the configuration for your Auth0 app is imported from `auth_config.json` and supplied to the plugin:
 
 ```js
-// src/main.js
-
 import Vue from "vue";
 import App from "./App.vue";
+import router from './router'
+import { domain, clientId } from "../auth_config.json";
 
 // Import the plugin here
-import AuthPlugin from "./plugins/auth";
+import { Auth0Plugin } from "./auth";
 
 // Install the authentication plugin here
-Vue.use(AuthPlugin);
+Vue.use(Auth0Plugin, {
+  domain,
+  clientId,
+  onRedirectCallback: appState => {
+    router.push(
+      appState && appState.targetUrl
+        ? appState.targetUrl
+        : window.location.pathname
+    );
+  }
+});
 
 Vue.config.productionTip = false;
 
 new Vue({
   router,
   render: h => h(App)
 }).$mount("#app");
-
 ```
 
-### Provide a Login Control
-
-Provide a component with controls for the user to log in and log out.
-
-::: note
-This example was created from a [Vue CLI](https://cli.vuejs.org/) template and uses Single-File Components.
-:::
-
-${snippet(meta.snippets.use)}
-
-The `@click` events on the **Log In** and **Log Out** buttons make the appropriate calls to the `AuthService` to allow the user to log in and log out. Notice that these buttons are conditionally hidden and shown depending on whether or not the user is currently authenticated.
-
-Also notice the use of `this.$auth` to access the `AuthService` instance. Login events can be handled by providing a `handleLoginEvent` method on any component.
-
-When the **Log In** button is clicked, the user will be redirected to login page.
-
-<%= include('../../_includes/_hosted_login_customization') %>
-
-When the application first starts up, a call to `renewTokens` is made that tries to reinitialize the user's login session, if it is detected that they should already be logged in. This would be the case, for example, if the user logged in and then refreshed the browser window.
+Notice that the configuration file created earlier has been imported and used to initialize the plugin. A custom redirect callback has also been supplied, which redirects the user to a protected route after they have authenticated (this is covered a bit later).
```

Comment:
> A custom redirect callback has also been supplied ...

Has it? See comment below RE: callback error.

Reply:
From a previous comment, I've added in a comment above the import which *should* make it more visible - what do you think?

Reply:
I think my confusion was around the wording:

> custom redirect callback ...

You're talking about a redirect after logging in here, correct? The redirect to the profile page works fine but I think "callback" used in this context is a little misleading (we typically use "callback" to indicate where the auth is processed which, in this case, is just the homepage).

Reply:
I see what you're saying. I've changed the wording to "handler" - how does that sound?